src/schema/search/SearchResolver.ts

```diff
@@ -0,0 +1,137 @@
+import { convertConnectionArgsToGravityArgs } from "lib/helpers"
+import { createPageCursors, pageToCursor } from "schema/fields/pagination"
+import { connectionFromArraySlice } from "graphql-relay"
+import { GraphQLResolveInfo, visit } from "graphql"
+import { ResolverContext } from "types/graphql"
+import { Searchable } from "schema/searchable"
+import { SearchableItem } from "schema/SearchableItem"
+
+export class SearchResolver {
+  private args: any
+  private context: ResolverContext
+  private info: GraphQLResolveInfo
+  private cachedEntityTypesToFetch: string[] | undefined
+
+  constructor(args: any, context: any, info: any) {
```

Comment:
You can DRY up the class declaration _and_ type the constructor parameters to ensure you’re receiving the correct data from the call-site by leveraging [“parameter properties”](https://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties).

Reply:
Also, with this larger refactor work and adding the indirection of using the args in a different place than where they are defined, I think it might be a good time to start typing those args both in the `GraphQLFieldConfig` and here on the constructor parameter.

Reply:
Just added the types but I was a bit unsure of how to DRY it up further. Happy to pair on that sometime!